src/control/server/ctl_storage_rpc_test.go

```diff
@@ -777,9 +760,33 @@ func TestStorageFormat(t *testing.T) {
 				t.Fatal(fmtErr)
 			}
 
-			results := []*StorageFormatResp{resp}
-			if diff := cmp.Diff(tc.expResults, results); diff != "" {
-				t.Fatalf("unexpected results: (-want, +got):\n%s\n", diff)
+			common.AssertEqual(t, len(tc.expResp.Crets), len(resp.Crets),
+				"number of controller results")
+			common.AssertEqual(t, len(tc.expResp.Mrets), len(resp.Mrets),
+				"number of mount results")
```

Comment:
I'm curious why you decided to split these out, and the comparisons below, instead of using cmp.Diff to compare resp to expResp.

Reply:
Me too.

Reply:
the order of the lists were not deterministic hence the test failed intermittently (as we are now running separate go-routines). open to better suggestions

Reply:
Might not be the direction you want to go in, but `cmp.Transformer()` might be something to consider, e.g. https://github.com/daos-stack/daos/blob/master/src/control/lib/control/rpc_test.go#L175